cli/connect/lua/eval_func_body.lua

```diff
@@ -25,4 +58,46 @@ for i=2,ret.n do
         ret[i] = box.NULL
     end
 end
+if not need_metainfo then
+    return yaml.encode({unpack(ret, 2, ret.n)})
+end
+
+if ret.n == 2 and type(ret[2]) == 'cdata' and ret[2] ~= nil and ret[2].format then
+    local ret_with_format = {
+        metadata = ret[2]:format(),
+        rows = { ret[2] },
+    }
+    if #ret_with_format.metadata > 0 then
+        ret[2] = ret_with_format
+    end
+end
+
+if ret.n == 2 and type(ret[2]) == 'table' and #ret[2] > 0 then
```

Comment:
The same about ret[3], ret[4] etc. Here we need to write more general code.

Reply:
Updated.